s2/encode.go

```diff
@@ -222,3 +222,22 @@ func (d *decoder) readUvarint() (x uint64) {
 	x, d.err = binary.ReadUvarint(d.r)
 	return
 }
+
+func (d *decoder) readFloat64Array(size int, buffers [][]byte) ([]byte, int) {
+	if d.err != nil {
+		return nil, 0
+	}
+
+	var buf []byte
+
+	for _, buffer := range buffers {
```

Comment:
instead of passing around [][]byte as buffer, can we create a new type as GeoBuffPool or something ? 

That can we move this logic of selecting a best-fit buffer to some method of that new type.

Thoughts ?